From 5dd17689526971c5ae12bc8398f34bd68cd0499e Mon Sep 17 00:00:00 2001
From: Douglas Anderson <dianders@chromium.org>
Date: Fri, 20 Oct 2023 14:06:53 -0700
Subject: [PATCH] r8152: Run the unload routine if we have errors during probe
Git-commit: 5dd17689526971c5ae12bc8398f34bd68cd0499e
References: git-fixes
Patch-mainline: v6.6

The rtl8152_probe() function lacks a call to the chip-specific
unload() routine when it sees an error in probe. Add it in to match
the cleanup code in rtl8152_disconnect().

Fixes: ac718b69301c ("net/usb: new driver for RTL8152")
Signed-off-by: Douglas Anderson <dianders@chromium.org>
Reviewed-by: Grant Grundler <grundler@chromium.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Oliver Neukum <oneukum@suse.com>

---
 drivers/net/usb/r8152.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/usb/r8152.c
+++ b/drivers/net/usb/r8152.c
@@ -5281,6 +5281,8 @@ static int rtl8152_probe(struct usb_inte
 
 out1:
 	netif_napi_del(&tp->napi);
+	if (tp->rtl_ops.unload)
+		tp->rtl_ops.unload(tp);
 	usb_set_intfdata(intf, NULL);
 out:
 	free_netdev(netdev);
